plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessions.php

```diff
@@ -0,0 +1,452 @@
+		return true;
+	}
+
+	/**
+	 * Check if a nonce is required for the route.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool False, Bearer token auth used instead.
+	 */
+	protected function requires_nonce( \WP_REST_Request $request ) {
+		// Should use `is_authorized` to validate Bearer token authentication.
+		return false;
+	}
+
+	/**
+	 * Handle the request and return a valid response for this endpoint.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response
+	 */
+	protected function get_route_post_response( \WP_REST_Request $request ) {
+
+		// Add items to cart.
+		$items = $request->get_param( 'items' );
+
+		foreach ( $items as $index => $item ) {
+			if ( ! is_numeric( $item['id'] ) ) {
+				return new \WP_REST_Response(
+					[
+						'type'    => 'invalid_request',
+						'code'    => 'invalid_product_id',
+						'message' => __( 'Product ID must be numeric.', 'woocommerce' ),
+						'param'   => '$.items[' . $index . '].id',
+					],
+					400
+				);
+			}
+
+			$product_id = (int) $item['id'];
+			$quantity   = (int) $item['quantity'];
+
+			try {
+				$this->cart_controller->add_to_cart(
+					[
+						'id'       => $product_id,
+						'quantity' => $quantity,
+					]
+				);
+			} catch ( \Exception $e ) {
+				// Map exception messages to protocol-compliant error responses.
+				$error_code    = 'invalid_product';
+				$error_message = $e->getMessage();
+
+				// Detect specific error types from exception message.
+				if ( strpos( $error_message, 'stock' ) !== false ) {
+					$error_code = 'out_of_stock';
+				} elseif ( strpos( $error_message, 'purchasable' ) !== false || strpos( $error_message, 'available' ) !== false ) {
+					$error_code = 'product_not_purchasable';
+				} elseif ( strpos( $error_message, 'not found' ) !== false || strpos( $error_message, 'does not exist' ) !== false ) {
+					$error_code = 'invalid_product';
+				}
+
+				return new \WP_REST_Response(
+					[
+						'type'    => 'invalid_request',
+						'code'    => $error_code,
+						'message' => $error_message,
+						'param'   => '$.items[' . $index . ']',
+					],
+					400
+				);
+			}
+		}
+
+		// Set buyer information.
+		$buyer = $request->get_param( 'buyer' );
+		if ( $buyer ) {
+			$this->set_buyer_data( $buyer );
+		}
+
+		// Set fulfillment address.
+		$address = $request->get_param( 'fulfillment_address' );
+		if ( $address ) {
+			$this->set_fulfillment_address( $address );
+		} else {
+			// Clear address when not provided (POST creates fresh session).
+			$this->clear_fulfillment_address();
+		}
+
+		// Set selected shipping method if provided.
+		$fulfillment_option_id = $request->get_param( 'fulfillment_option_id' );
+		if ( $fulfillment_option_id ) {
+			WC()->session->set( 'chosen_shipping_methods', array( $fulfillment_option_id ) );
+		}
+
+		// Calculate totals after shipping method is set.
+		WC()->cart->calculate_totals();
```

Comment:
Should we also calculate taxes before we calculate the totals?

Reply:
Looking into this, taxes and fees are calculated as part of calculate_totals. So, I removed the calls to calculate shipping from your other comment.